### PR TITLE
 Dev: bootstrap: Generate the public key on the remote if it does not exist

### PR DESCRIPTION
**Pull request by liangxin1300**

### CODE REVIEW

**Comment by nicholasyang2022** [comment]
Does it do any harm to walk through all the init process? Splitting the implementation makes the code more complex.

**Comment by liangxin1300** [comment]
So, I think there are two solutions:
1. skip non hacluster part
2. generate public key based on private key, on peer node

**Comment by liangxin1300** [comment]
> So, I think there are two solutions:
> 
>     1. skip non hacluster part
> 
>     2. generate public key based on private key, on peer node
> 
> 
> Which one do you prefer?

@nicholasyang2022 ,I already changed as the second solution

File: crmsh/bootstrap.py [hunk 1]
```diff
@@ -1049,6 +1049,20 @@ def generate_ssh_key_pair_on_remote(
     )
     if result.returncode != 0:
         raise ValueError(codecs.decode(result.stdout, 'utf-8', 'replace'))
+
+    cmd = '''
+[ -f ~/.ssh/id_rsa.pub ] || ssh-keygen -y -f ~/.ssh/id_rsa > ~/.ssh/id_rsa.pub
+'''
```
Comment (nicholasyang2022) [inline crmsh/bootstrap.py:1055]: This can be append to the previous `cmd` and called with one ssh session, making it faster.

Comment (liangxin1300) [reply crmsh/bootstrap.py:1043]: Changed

**Review by nicholasyang2022 — APPROVED** [review]

**Pull request merged**